rasa/core/actions/action.py

```diff
@@ -940,3 +949,141 @@ def name(self) -> Text:
     def __init__(self) -> None:
         """Initializes action default ask rephrase."""
         super().__init__("utter_ask_rephrase", silent_fail=True)
+
+
+class ActionExtractSlots(Action):
+    """Silent action that runs after each user turn.
+
+    Sets slots to extracted values from user message
+    according to assigned slot mappings.
+    """
+
+    def __init__(self, action_endpoint: Optional[EndpointConfig]) -> None:
+        """Initializes default action extract slots."""
+        self.action_endpoint = action_endpoint
+
+    def name(self) -> Text:
+        """Returns action_extract_slots name."""
+        return ACTION_EXTRACT_SLOTS
+
+    async def run(
+        self,
+        output_channel: "OutputChannel",
+        nlg: "NaturalLanguageGenerator",
+        tracker: "DialogueStateTracker",
+        domain: "Domain",
+    ) -> List[Event]:
+        """Runs action. Please see parent class for the full docstring."""
+        slot_events: List[Event] = []
+
+        default_slots = {
+            rasa.shared.core.constants.REQUESTED_SLOT,
+            rasa.shared.core.constants.SESSION_START_METADATA_SLOT,
+            rasa.shared.core.constants.SLOT_LISTED_ITEMS,
+            rasa.shared.core.constants.SLOT_LAST_OBJECT,
+            rasa.shared.core.constants.SLOT_LAST_OBJECT_TYPE,
+        }
+        user_slots = [slot for slot in domain.slots if slot.name not in default_slots]
+
+        for slot in user_slots:
+            for mapping in slot.mappings:
+                # skip to the next mapping because a mapping with conditions
+                # is applicable only within the context of an active loop
+                if mapping.get("conditions"):
+                    continue
+
+                intent_is_desired = SlotMapping.intent_is_desired(
+                    mapping, tracker, domain
+                )
+
+                should_fill_trigger_slot = (
+                    mapping["type"] == str(SlotMapping.FROM_TRIGGER_INTENT)
+                    and intent_is_desired
+                )
+
+                if should_fill_trigger_slot:
+                    # skip, only applicable in an active loop
+                    continue
+
+                should_fill_custom_slot = (
+                    mapping["type"] == str(SlotMapping.CUSTOM) and intent_is_desired
+                )
+
+                should_fill_entity_slot = (
+                    mapping["type"] == str(SlotMapping.FROM_ENTITY)
+                    and intent_is_desired
+                    and SlotMapping.entity_is_desired(mapping, tracker,)
+                )
+
+                should_fill_intent_slot = (
+                    mapping["type"] == str(SlotMapping.FROM_INTENT)
+                    and intent_is_desired
+                )
+
+                should_fill_text_slot = (
+                    mapping["type"] == str(SlotMapping.FROM_TEXT) and intent_is_desired
+                )
```

Comment:
It doesn't look like we really need to store all that (`should_fill_text_slot`, `should_fill_intent_slot`, etc) and have separate variables for them.

Reply:
I actually thought it's more readable like this and also consistent to how slots get extracted in `FormAction` too 🤔

Reply:
Well, that might be true. Let's skip it then.